src/workerd/api/tests/js-rpc-test.js

```diff
@@ -0,0 +1,180 @@
+import assert from 'node:assert';
+import {StatelessService,DurableObject} from 'cloudflare:entrypoints';
+
+export let nonClass = {
+  async noArgs(x, env, ctx) {
+    return (x === undefined) ? (env.twelve + 1) : "param not undefined?";
+  },
+
+  async oneArg(i, env, ctx) {
+    return env.twelve * i;
+  },
+
+  async twoArgs(i, env, ctx, j) {
+    return i * j + env.twelve;
+  },
+}
+
+export class MyService extends StatelessService {
+  #env;
+
+  constructor(ctx, env) {
+    super(ctx, env);
+    this.#env = env;
+
+    // This shouldn't be callable!
+    this.instanceMethod = () => {
+      return "nope";
+    }
+  }
+
+  async noArgsMethod(x) {
+    return (x === undefined) ? (this.#env.twelve + 1) : "param not undefined?";
+  }
+
+  async oneArgMethod(i) {
+    return this.#env.twelve * i;
+  }
+
+  async twoArgsMethod(i, j) {
+    return i * j + this.#env.twelve;
+  }
+
+  async fetch(req, x) {
+    assert.strictEqual(x, undefined);
+    return new Response("method = " + req.method + ", url = " + req.url);
+  }
+}
+
+export class MyActor extends DurableObject {
+  #counter = 0;
+
+  constructor(state, env) {
+    super(state, env);
```

Comment:
Is the superclass making properties with these names? I feel like it should for convenience, and because there's no risk of exposing them over RPC accidentally.

Either way, for this class, you can omit the constructor entirely

Reply:
I think we still don't want to:
* We _are_ going to allow exposing properties as long as they are on the class, not instance properties. So we'd have to make special exceptions for these which is weird.
* I don't like them being automatically public to local code, even if they aren't exposed over RPC.
* I am not sure the names `state` and `env` (and `ctx` for stateless) are the best names. We've always used them in docs but nothing in the API actually mandates these specific names today. I'd rather not have to think about this.